routes/index.js

```diff
@@ -46,23 +46,28 @@ router.post('/', upload.single('avatar'), function (req, res, next){
     var lastName = req.body.lastName;
     var number = req.body.number;
     var country = req.body.country;
-    ext = req.file.mimetype.replace("image/", ".");
-    var avatar = req.file.path;
-    var newContact = {firstName: firstName, lastName: lastName, number: number, country: country, avatar: avatar};
-
-    if(ext === '.png' || ext === '.jpg' || ext === '.jpeg') {
-        Contact.create(newContact, function(err, newlyCreated){
-            if(err) {
-                console.log(err);
-            }
-            else {
-                res.redirect('/');
-            }
-        });
-    }
-    else {
-        res.send('File type not supported, use .jpg or .png. Click back to try again');
-    }
+    var avatar = req.file
+
+    var dUri = new Datauri()
+    dUri.format(path.extname(req.file.originalname).toString(), req.file.buffer)
+    cloudinary.uploader.upload(dUri.content, function(results) {
```

Comment:
First we upload it to cloudinary, and then we wait for the `results` argument. It has many useful properties for us, but the one we really care about is `results.url` or `results.secure_url` if you want https.